custom/enikshay/expressions.py

```diff
@@ -119,6 +121,113 @@ def concatenate_strings_expression(spec, context):
     return wrapped
 
 
+class ReferralExpressionBase(JsonObject):
+    person_id_expression = DefaultProperty(required=True)
+
+    def configure(self, person_id_expression):
+        self._person_id_expression = person_id_expression
+
+    def __call__(self, item, context=None):
+        person_id = self._person_id_expression(item, context)
+        domain = context.root_doc['domain']
+        if not person_id:
+            return None
+
+        referral = get_open_referral_case_from_person(domain, person_id)
```

Comment:
Are these going to be called multiple times or exist in multiple data sources? It may be worth caching them